### ISSUE TITLE
Go template "exec" function fails silently when used in a pipeline

### ISSUE THREAD

**Issue by elemental-lf** [roboll/helmfile#300]
With 0.33.0 the "exec" function fails silently when the exec fails and the function is used as part of a pipeline.

The following Go template code fails silently and reports success even when ../scripts/generate-osds.py isn't found:
``
{{ readFile (printf "../values/%s/ceph-osds.yaml" (requiredEnv "PLATFORM_ENV")) | exec "../scripts/generate-osds.py" (list) }}
``
If no pipeline is involved and the "exec" function is used just on it's own, I get the expected error message and helmfile terminates.

**Comment by mumoshu** [comment]
@elemental-lf Thanks for the report! I have never realized about the behavior of golang text/template, and I do want to fix it.

What we're doing here is that make the template func return `(string, error)` so I believe that it's all up to how text/template deals with an error in pipelines. Maybe do `panic` instead of returning an error from within template funcs makes a difference? 😈Indeed we need more inevstigation..

**Comment by elemental-lf** [comment]
@mumoshu I wonder, how does Helm handle this case? I think we should align to it. I'd be surprised if it didn't fail in some way, but the implementation might be worth a look.

**Comment by mumoshu** [comment]
@elemental-lf Is there any chance that your script is returning a successful exit code on the pipe input?

I've tried to create a minimal example to reproduce the issue, but had no luck. A template function seems to be able to return an error to make the template execution fail. PTAL on [my code on the playground](https://play.golang.org/p/UhW-KvmVVh9).

**Comment by mumoshu** [comment]
> You were testing the case where the script returned a non-zero exit code, right?

Exactly. I'll confirm how it behaves on the inaccessible file case. Thanks!
